### PR TITLE
solana: Added client-side tx rebroadcast logic

### PR DESCRIPTION
**Pull request by kev1n-peters**
Implements the suggestions from here: https://docs.triton.one/chains/solana/sending-txs

### CODE REVIEW

File: platforms/solana/src/platform.ts [hunk 1]
```diff
@@ -176,41 +178,66 @@ export class SolanaPlatform<N extends Network>
...
+  static async sendTxWithRetry(
+    rpc: Connection,
+    tx: SignedTx,
+    retryInterval = 5000,
+  ): Promise<{
+    signature: string;
+    response: RpcResponseAndContext<SignatureResult>;
+  }> {
+    const sendOptions: SendOptions = {
+      skipPreflight: true, // TODO: need to make sure tx was simulated just before sending
```
Comment (kev1n-peters) [inline platforms/solana/src/platform.ts:206]: We are skipping preflight here (important), but this is different than the previous behavior. Previously the call to `sendRawTransaction` would set `skipPreflight: false` by default. So now we either need to simulate the transaction here just before sending it or expect the caller to have done it. Callers already simulate the tx to get the compute budget, so simulating it again here would be redundant. Maybe we can add a flag to `determineComputeBudget` to throw when simulation fails. Not sure what's best here

Comment (kev1n-peters) [reply platforms/solana/src/platform.ts:206]: Actually, I think we could set `skipPreflight: false` for the first call to `sendRawTransaction` so it throws if simulation fails and then set it to `true` on any subsequent calls.

File: platforms/solana/src/platform.ts [hunk 2]
```diff
@@ -176,41 +178,74 @@ export class SolanaPlatform<N extends Network>
...
+    response: RpcResponseAndContext<SignatureResult>;
+  }> {
+    const commitment = sendOpts.preflightCommitment ?? rpc.commitment;
+    const signature = await rpc.sendRawTransaction(tx, {
+      ...sendOpts,
+      skipPreflight: false, // The first send should not skip preflight to catch any errors
+      maxRetries: 0,
+      preflightCommitment: commitment,
+    });
+    // TODO: Use the lastValidBlockHeight that corresponds to the blockhash used in the transaction.
```
Comment (barnjamin) [inline platforms/solana/src/platform.ts:213]: should we do this TODO?

File: platforms/solana/src/platform.ts [hunk 3]
```diff
@@ -176,41 +178,74 @@ export class SolanaPlatform<N extends Network>
...
+    sendOpts: SendOptions = {},
+    retryInterval = 5000,
+  ): Promise<{
+    signature: string;
+    response: RpcResponseAndContext<SignatureResult>;
+  }> {
+    const commitment = sendOpts.preflightCommitment ?? rpc.commitment;
+    const signature = await rpc.sendRawTransaction(tx, {
+      ...sendOpts,
+      skipPreflight: false, // The first send should not skip preflight to catch any errors
```
Comment (barnjamin) [inline platforms/solana/src/platform.ts:209]: what is the default? assuming false, can this be removed?

Comment (kev1n-peters) [reply platforms/solana/src/platform.ts:209]: Yeah it's false by default, but I would prefer to be explicit here with the accompanying comment

File: platforms/solana/src/platform.ts (hunk 2 above)
Comment (kev1n-peters) [reply platforms/solana/src/platform.ts:213]: I think we should, but there's no way to get the `lastValidBlockHeight` from a `VersionedTransaction`. I think we're going to have to change the `SignedTx` type for Solana from `Buffer` to { data: buffer, recentBlockhash, lastValidBlockHeight }`. The `recentBlockhash` and `lastValidBlockHeight` are needed in the call to `confirmTransaction`.

**Review by barnjamin — APPROVED** [review]

File: platforms/solana/src/platform.ts (hunk 2 above)
Comment (kev1n-peters) [reply platforms/solana/src/platform.ts:213]: Will revisit in a future PR. Not using the `recentBlockhash` and `lastValidBlockHeight` from the transaction object is how it worked prior to the changes in this PR.

**Pull request merged**